### PR TITLE
Fix video mirroring issue

### LINKED ISSUE #1804: Video mirroring
### Describe the bug

Upload a video, and the output video is a mirror image.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

```python
import gradio as gr

def video_test(video):
    return video

inputs_video_p5 = gr.Video(format="mp4", source="upload", label="原始视频")  # webcam
outputs_webcam_p5 = gr.Video(format='mp4', label="检测视频")

gyd = gr.Interface(
    fn=video_test,
    inputs=[inputs_video_p5],
    outputs=[outputs_webcam_p5],
)
gyd.launch(
    inbrowser=True,
)
```

### Screenshot

ksnip_20220716-145557

### Logs

```shell
see image above
```

### System Info

```shell
gradio 3.0.25
ubuntu 20.04
chrome
```

### Severity

annoying

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description

Fixes #1804

There are two problems that I found:

1) When source="upload", the front-end was flipping videos since `mirror_webcam` is always set to True in the config. Regardless of whether the source is a webcam. This is the problem causing #1804. 
2) When the source is webcam and the `format` is specified (not None), the video was not flipped in the backend. There isn't an issue about this filed I think.

### Issue 1 Test code
```python
import gradio as gr

def video_test(video):
    return video

inputs_video_p5 = gr.Video(format="mp4", source="upload", label="原始视频")  # webcam
outputs_webcam_p5 = gr.Video(format='mp4', label="检测视频")

gyd = gr.Interface(
    fn=video_test,
    inputs=[inputs_video_p5],
    outputs=[outputs_webcam_p5],
)
gyd.launch(
    inbrowser=True,
)
```
no_flip_if_upload

### Issue 2 Test Code
```python
import gradio as gr

def video_test(video):
    return video

inputs_video_p5 = gr.Video(format="mp4", source="webcam", label="原始视频")  # webcam
outputs_webcam_p5 = gr.Video(format='mp4', label="检测视频")

gyd = gr.Interface(
    fn=video_test,
    inputs=[inputs_video_p5],
    outputs=[outputs_webcam_p5],
)
gyd.launch(
    inbrowser=True,
)
```

video_mirrored_if_format_change

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

File: gradio/components.py [hunk 1]
```diff
@@ -1640,25 +1640,28 @@ def preprocess(self, x: Dict[str, str] | None) -> str | None:
...
-            return output_file_name
-        elif self.source == "webcam" and self.mirror_webcam is True:
-            path = Path(file_name)
-            output_file_name = str(path.with_stem(f"{path.stem}_flip"))
+        file_name = Path(file.name)
+        uploaded_format = file_name.suffix.replace(".", "")
+
+        modify_format = self.format is not None and uploaded_format != self.format
+        flip = self.source == "webcam" and self.mirror_webcam is True
+        if modify_format or flip:
```
Comment (freddyaboulton) [inline gradio/components.py:1648]: Consolidating the two  FFmpeg calls into one.

**Comment by pngwn** [comment]
Code looks good but I could never reproduce this, so will leave for someone else to test. It works fine for me, but it always did.

File: gradio/components.py [hunk 2]
```diff
@@ -1640,25 +1640,28 @@ def preprocess(self, x: Dict[str, str] | None) -> str | None:
...
-            ff.run()
-            return output_file_name
-        elif self.source == "webcam" and self.mirror_webcam is True:
-            path = Path(file_name)
-            output_file_name = str(path.with_stem(f"{path.stem}_flip"))
+        file_name = Path(file.name)
+        uploaded_format = file_name.suffix.replace(".", "")
+
+        modify_format = self.format is not None and uploaded_format != self.format
+        flip = self.source == "webcam" and self.mirror_webcam is True
```
Comment (abidlabs) [inline gradio/components.py:1647]: Nit:
```suggestion
        flip = self.source == "webcam" and self.mirror_webcam
```

File: gradio/components.py [hunk 3]
```diff
@@ -1640,25 +1640,28 @@ def preprocess(self, x: Dict[str, str] | None) -> str | None:
...
+        if modify_format or flip:
+            format = f".{self.format if modify_format else uploaded_format}"
+            output_options = ["-vf", "hflip", "-c:a", "copy"] if flip else None
+            flip_suffix = "_flip" if flip else ""
+            output_file_name = str(
+                file_name.with_name(
+                    f"{file_name.name.replace(file_name.suffix, '')}{flip_suffix}{format}"
```
Comment (abidlabs) [inline gradio/components.py:1654]: Would this work?
```suggestion
                    f"{file_name.stem}{flip_suffix}{format}"
```

**Comment by abidlabs** [comment]
Looks awesome @freddyaboulton!

**Review by abidlabs — APPROVED** [review]

File: gradio/components.py (hunk 3 above)
Comment (freddyaboulton) [reply gradio/components.py]: Yes!

**Comment by freddyaboulton** [comment]
Thanks for the review all!

**Pull request merged**